app/lib/Server.scala

```diff
@@ -0,0 +1,252 @@
+package lib
+
+
+import scala.util.Try
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import cats.implicits._
+import cats.data.EitherT
+
+import play.api.libs.json._
+import com.github.nscala_time.time.Imports._
+
+import lib.Data._
+import lib.Seats._
+import lib.DBInterface
+
+class Server(dbInterface: DBInterface) {
+
+
+  // Response preparation -------------------------------------------------------------------------
+  def screeningInfoBasic(info: ScreeningInfo): JsObject = {
+    val startHour   = info.start.toLocalTime.getHourOfDay
+    val startMinute = info.start.toLocalTime.getMinuteOfHour
+
+    Json.obj(
+      "screeningId" -> objectIdToString(info.id),
+      "title"       -> info.title,
+      "start-time"  -> f"$startHour%02d:$startMinute%02d",
+      "duration"    -> info.duration.toStandardMinutes.getMinutes,
+    )
+  }
+
+  def screeningInfo(info: ScreeningInfo): EitherT[Future, Error, JsObject] = for {
+
+    availableSeats <- getAvailableSeats(info.id, info.room)
+
+    basicInfo = screeningInfoBasic(info)
+
+    date = DateTimeFormat.forPattern("dd-MM-yyyy").print(info.start)
+
+    result = (
+      basicInfo
+      + ("date" -> Json.toJson(date))
+      + ("room" -> Json.toJson(info.room))
+      + ("rows" -> Json.toJson(availableSeats.dim.numRows))
+      + ("seats-per-row" -> Json.toJson(availableSeats.dim.numColumns))
+      + ("available-seats" -> Json.toJson(availableSeats.seats))
+    )
+
+  } yield result
+
+
+  // Request body processing ----------------------------------------------------------------------
+  def processReservationData(
+    screeningId:  ScreeningId,
+    body:         JsValue,
+  ): Option[Reservation] = body match {
+
+    case JsObject(obj) => for {
+      jsonName    <- obj.get("name")
```

Comment:
I don't know of any such library. Since I'm new to scala, I already had to find and learn all the other frameworks and I found it easier to write the json extraction myself, than to search for and figure aout another library.
If I wanted to use the same structure somewhere else, I would refactor this method so that I would't repeat the code or I would just create another module for processing json and put this method there.